### DISCUSSION TITLE
Crazyflie2.1 crashes. Warning: Low kalman prediction rate

### DISCUSSION THREAD

**Discussion by hari-2309** [bitcraze/discussions#933] [Q&A]
Hello,

I recently assembled a fresh crazyflie2.1 

The drone crashes every time I try to fly it. I am new to working with drones, so detailed instructions will be helpful.

The propeller test results and task dump results for the drone are given below:

HEALTH: Acc noise floor variance X+Y:0.002684, (Z:0.001472)
HEALTH: Motor M1 variance X+Y: 0.71 (Z:3.62), voltage sag:0.21
HEALTH: Motor M2 variance X+Y: 0.52 (Z:0.91), voltage sag:0.20
HEALTH: Motor M3 variance X+Y: 1.86 (Z:9.49), voltage sag:0.21
HEALTH: Motor M4 variance X+Y: 0.83 (Z:2.30), voltage sag:0.20
ESTKALMAN: WARNING: Kalman prediction rate low (83)
SYSLOAD: Task dump
SYSLOAD: Load	Stack left	Name
SYSLOAD: 3.73 	179 	Tmr Svc
SYSLOAD: 63.57 	127 	IDLE
SYSLOAD: 0.0 	75 	PWRMGNT
SYSLOAD: 1.69 	213 	CRTP-RX
SYSLOAD: 6.65 	148 	SENSORS
SYSLOAD: 2.19 	75 	ZRANGER2
SYSLOAD: 2.54 	191 	FLOW
SYSLOAD: 0.0 	115 	PLATFORM-
SYSLOAD: 0.0 	251 	LOG
SYSLOAD: 0.0 	223 	MEM
SYSLOAD: 0.0 	258 	PARAM
SYSLOAD: 0.07 	157 	SYSTEM
SYSLOAD: 0.05 	61 	CRTP-TX
SYSLOAD: 2.40 	175 	SYSLINK
SYSLOAD: 2.90 	91 	LEDSEQCMD
SYSLOAD: 1.15 	107 	CRTP-SRV
SYSLOAD: 4.51 	308 	STABILIZE
SYSLOAD: 8.48 	307 	KALMAN
SYSLOAD: 0.0 	113 	USBLINK
SYSLOAD: 0.0 	255 	CMDHL

**Comment by knmcguire** [comment]
Hi! We need a bit more information. 

The kalman prediction rate low is a common warning so it probably doesn't have anything to do with it.
Currently we should look at other things than the taskdump first:

- Have you updated the crazyflie to the latest release firmware?
- which decks are you flying with, or is it just in manual flight?
- Could you share the console tab output as well?
- What do you mean with crashing? Is the crazyflie unable to fly? Does it restart and flash the red LED?

**Reply by hari-2309** [reply]
Hi,

1. Yes, I updated the crazyflie to the latest version.
2. The only additional deck added is flowdeck. 
3. Console tab output:
SYS: ----------------------------
SYS: Crazyflie 2.1 is up and running!
SYS: Build 0:f83e4269f59b (2023.07) CLEAN
SYS: I am 0x313232313232471400300032 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 1 deck(s) found
DECK_CORE: Calling INIT on driver bcFlow2 for deck 0
ZR2: Z-down sensor [OK]
PMW: Motion chip id: 0x49:0xB6
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Kalman (2) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2023.02
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
DECK_CORE: Deck 0 test [OK].
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 17608 bytes
STAB: Starting stabilizer loop
ESTKALMAN: State out of bounds, resetting
SUP: Can not fly
SUP: Ready to fly

4. The crazyflie takes off, drifts and then abruptly crashes down. This is happening in position hold mode.

**Reply by knmcguire** [reply]
Ah oke, and it doesn't show the red LED flashing 3 times quickly?

Just to eliminate any controller issues, can you try out the takeoff button in 'Command based flight control' in the flightcontrol tab? https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/flightcontrol_tab/

also make sure that you are flying above an floor with some texture.

**Reply by hari-2309** [reply]
The red led doesn't flash 3 times quickly. 

I tried the takeoff option and the first time the drone flipped down. The second time it took of at a very high speed and then abruptly crashed down.

**Reply by knmcguire** [reply]
that is indeed very strange. It perhaps seems like an issue with the motors.

Some other things you can try:

* Are you able to control the drone manually without position control (remove the flowdeck)?
* Are any of the motors badly connected? That the white connector seems shifted, the plug not mounted in properly, or one of the wires that looked a bit messed up? If you have spare motors you can try replacing them (maybe one by one).
* What is the state of the propellers? Do they look banged up? How about the propeller mounts?
* Do you have another crazyflie to try out? How does that one perform?

Perhaps if it is possible to show a video of the flight and crash?

**Reply by hari-2309** [reply]
Hi,

I tried with another crazyflie and it worked fine.
For the drone that is crashing, the propellers and connections seem to be fine. The propeller mounts appear straight too.

**Reply by knmcguire** [reply]
Alright good to know that you have another one:

* Does the same flowdeck from the crashing crazyflie work on your spare?
* are you able to control the crashing crazyflie [manually in attitude control](https://www.bitcraze.io/documentation/tutorials/getting-started-with-crazyflie-2-x/#flying)? 
* If you want, you can try to also replace the motors of the crashed crazyflie with your spare, or if you have any spare motors. I kind of suspect there is on broken motor on that crashing crazyflie.
